subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/deps/ClassSetAnalysisData.java

```diff
@@ -101,12 +124,16 @@ public ClassSetAnalysisData read(Decoder decoder) throws Exception {
 
             String fullRebuildCause = decoder.readNullableString();
 
-            return new ClassSetAnalysisData(dependentsBuilder.build(), classesToConstantsBuilder.build(), classNameToChildren.build(), fullRebuildCause);
+            return new ClassSetAnalysisData(classes.build(), dependentsBuilder.build(), classesToConstantsBuilder.build(), classNameToChildren.build(), fullRebuildCause);
         }
 
         @Override
         public void write(Encoder encoder, ClassSetAnalysisData value) throws Exception {
             Map<String, Integer> classNameMap = new HashMap<String, Integer>();
+            encoder.writeSmallInt(value.classes.size());
+            for (String clazz : value.classes) {
+                writeClassName(clazz, classNameMap, encoder);
+            }
```

Comment:
This is stored in a version-specific cache so we don't have to worry about older versions reading/writing, right?

Reply:
That's right.